completers/tox_completer/cmd/exec.go

```diff
@@ -14,30 +16,40 @@ var execCmd = &cobra.Command{
-	execCmd.Flags().String("no-provision", "", "do not perform provision, but fail and if a path was provided write provision metadata as JSON to it (default: False)")
-	execCmd.Flags().Bool("no-recreate-pkg", false, "if recreate is set do not recreate packaging tox environment(s) (default: False)")
-	execCmd.Flags().Bool("no-recreate-provision", false, "if recreate is set do not recreate provision tox environment (default: False)")
+	execCmd.Flags().StringS("e", "e", "", "environment to run (default: py)")
+	execCmd.Flags().String("skip-env", "", "exclude all environments selected that match this regular expression (default: '')")
+	execCmd.Flags().StringP("skip-missing-interpreters", "s", "config", "don't fail tests for missing interpreters: {config,true,false} choice (default: config)")
 	execCmd.Flags().BoolP("notest", "n", false, "do not run the test commands (default: False)")
-	execCmd.Flags().StringP("pkg-only,", "b", "", "")
-	execCmd.Flags().BoolP("quiet", "q", false, "decrease verbosity (default: 0)")
-	execCmd.Flags().BoolP("recreate", "r", false, "recreate the tox environments (default: False)")
-	execCmd.Flags().String("result-json", "", "write a JSON file with detailed information about all commands and results involved (default: None)")
-	execCmd.Flags().String("root", "", "project root directory (if not specified will be the folder of the config file) (default: None)")
-	execCmd.Flags().String("runner", "", "the tox run engine to use when not explicitly stated in tox env configuration (default: virtualenv)")
-	execCmd.Flags().String("skip-env", "", "exclude all environments selected that match this regular expression (default: )")
+
+	// these two are the same flag; there are two long flags for this
+	execCmd.Flags().BoolP("pkg-only,", "b", false, "only perform the packaging activity")
+	execCmd.Flags().Bool("sdistonly", false, "only perform the packaging activity")
+
+	execCmd.Flags().String("installpkg", "", "use specified package for installation into venv, instead of packaging the project (default: None)")
+	execCmd.Flags().Bool("develop", false, "install package in development mode (default: False)")
+	execCmd.Flags().Bool("no-recreate-pkg", false, "if recreate is set do not recreate packaging tox environment(s) (default: False)")
 	execCmd.Flags().Bool("skip-pkg-install", false, "skip package installation for this run (default: False)")
-	execCmd.Flags().BoolP("verbose", "v", false, "increase verbosity (default: 2)")
-	execCmd.Flags().Bool("version", false, "show program's and plugins version number and exit")
-	execCmd.Flags().String("workdir", "", "tox working directory (if not specified will be the folder of the config file) (default: None)")
+
+	carapace.Gen(execCmd).FlagCompletion(carapace.ActionMap{
+		"e":  tox.ActionEnvironments(),
+	})
+
+	carapace.Gen(execCmd).PositionalCompletion(
+		carapace.ActionValues("--"),
+	)
+
+	// TODO: how to get it to complete after `--`?
```

Comment:
[DashCompletion](https://carapace-sh.github.io/carapace/carapace/gen/dashCompletion.html) and [DashAnyCompletion](https://carapace-sh.github.io/carapace/carapace/gen/dashAnyCompletion.html).
`Context.Args` is reset for these so Actions can be used just as in positional completion.
If you need to continue the positional completion there is [ActionPositional](https://carapace-sh.github.io/carapace/carapace/defaultActions/actionPositional.html).

Reply:
Thanks! :)